trestle/core/parser.py

```diff
@@ -122,12 +121,11 @@ def parse_file(file_name: str, model_name: str):
     if file_name is None:
         raise TrestleError('file_name is required')
 
-    with open(file_name) as f:
-        data = yaml.load(f, yaml.FullLoader)
-        rkey = root_key(data)
-        if model_name is None:
-            model_name = to_full_model_name(rkey)
-        return parse_dict(data[rkey], model_name)
+    data = fs.load_file(file_name)
```

Comment:
Created issue #62 to have a look at this file.

I also wondering whether there are any benefits in using the inbuilt parsers from pydantic.

Reply:
I'm going to make this as resolved for now and to be deal with with issues relating to trestle.core.parser